File: Mooga/src/de/lumpn/mooga/Genome.java
package de.lumpn.mooga;

import java.util.Random;
import de.lumpn.util.Pair;

public interface Genome {

	/**
	 * Cross two genomes producing two offsprings.
	 */
	Pair<Genome> crossover(Genome other, Random random);

	/**
	 * Mutate genome returning the mutation.
	 */
	Genome mutate(Random random);
}

File: csharp/InfiniteZelda/Mooga/Genome.cs
using System;
using System.Collections.Generic;
using Lumpn.Utils;

namespace Lumpn.Mooga
{
    public interface Genome
    {
        /// Cross two genomes producing two offsprings.
        IEnumerable<Genome> Crossover(Genome other, RandomNumberGenerator random);

        /// Mutate genome returning the mutated copy.
        Genome Mutate(RandomNumberGenerator random);
    }
}
